layers/meta-balena-fsl-arm/recipes-core/images/resin-image.inc

```diff
@@ -75,3 +75,34 @@ IMAGE_CMD_resinos-img_append_solidrun-imx6 () {
     dd if=${DEPLOY_DIR_IMAGE}/${SPL_BINARY} of=${RESIN_RAW_IMG} conv=notrunc seek=1 bs=1K
     dd if=${DEPLOY_DIR_IMAGE}/u-boot-${MACHINE}.${UBOOT_SUFFIX} of=${RESIN_RAW_IMG} conv=notrunc seek=69 bs=1K
 }
+
+
+IMAGE_FSTYPES_append_nitrogen8mm = " resinos-img"
+RESIN_IMAGE_BOOTLOADER_nitrogen8mm = "u-boot"
+RESIN_BOOT_PARTITION_FILES_nitrogen8mm = " \
+    ${KERNEL_IMAGETYPE}${KERNEL_INITRAMFS}-${MACHINE}.bin:/${KERNEL_IMAGETYPE} \
+    imx8mm-${MACHINE}.dtb:/imx8mm-${MACHINE}.dtb \
+    boot.scr-${MACHINE}:/boot.scr \
+    imx-boot:/imx-boot \
+"
+
+IMAGE_INSTALL_append_nitrogen8mm = " \
+	imx-boot \
+	firmware-imx \
+	kernel-module-imx-gpu-viv \
+	kernel-module-qcacld \
+	linux-firmware-bdsdmac \
+	u-boot-script-boundary \
+"
+
+# zImage is available for up to imx7 only
+# so need to ensure boot partition is
+# large enough to accomodate future Image
+# growth
+RESIN_BOOT_SIZE_nitrogen8mm = "80960"
```

Comment:
aarch64 uses Image, zImage is for 32 bits. So probably won't be seeing zImage for this board anytime soon.
What is the Image size in your case?